pandas/core/series.py

```diff
@@ -1250,6 +1250,10 @@ def setitem(key, value):
     def _set_with_engine(self, key, value):
         values = self._values
         try:
+            if is_extension_array_dtype(values):
+                # The cython indexing routines do not support ExtensionArrays.
+                # Defer to the next setting routine.
+                raise KeyError
```

Comment:
can this go outside the try/except and instead of raising just copy line 1260?  it's one-liner either way